# Time Series Analysis of Daily Minimum Temperatures

In [ ]:
import pandas as pd

In [2]:
df = pd.read_csv("csv2/daily-min-temperatures.csv")
print(df.head(5))

         Date  Temp
0  1981-01-01  20.7
1  1981-01-02  17.9
2  1981-01-03  18.8
3  1981-01-04  14.6
4  1981-01-05  15.8


In [3]:
len(df)

3650

## Dataset Description

The dataset used in this analysis is the **Daily Minimum Temperatures** dataset, obtained from the following public repository:

 https://github.com/jbrownlee/Datasets

The dataset contains daily observations with two variables:

- **Date**: the date of the observation  
- **Temp**: the recorded minimum temperature  

The data spans multiple years and is structured as a univariate time series.

---

## Why this dataset?

This dataset was selected for several reasons:

- It is publicly available and widely used in time series learning contexts.
- It contains a relatively large number of observations (3650), which makes it suitable for statistical modeling.
- The dataset is simple and clean, allowing the analysis to focus on time series methods rather than data preprocessing.
- It represents a real-world time series with natural variability.

---

## Objective of the analysis

The objective of this analysis is to explore the temporal structure of the data and understand its behavior over time.

In particular, the analysis will focus on:

- Visualizing the time series  
- Evaluating stationarity  
- Analyzing autocorrelation patterns (ACF and PACF)  
- Identifying suitable models for time series forecasting  

The goal is to apply classical time series techniques and evaluate their performance on this dataset.

In [4]:
# Shape of the dataset
print("\nShape of the dataset (rows, columns):")
print(df.shape)



Shape of the dataset (rows, columns):
(3650, 2)


In [5]:
# Column names
print("\nColumn names:")
print(df.columns)


Column names:
Index(['Date', 'Temp'], dtype='str')


In [6]:
# Data types and general information
print("\nDataset info:")
print(df.info())


Dataset info:
<class 'pandas.DataFrame'>
RangeIndex: 3650 entries, 0 to 3649
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    3650 non-null   str    
 1   Temp    3650 non-null   float64
dtypes: float64(1), str(1)
memory usage: 57.2 KB
None


In [7]:
# Missing values
print("\nMissing values per column:")
print(df.isnull().sum())



Missing values per column:
Date    0
Temp    0
dtype: int64


In [8]:
# Check for duplicated rows
print("\nNumber of duplicated rows:")
print(df.duplicated().sum())



Number of duplicated rows:
0


In [9]:
# Convert Date to datetime
df["Date"] = pd.to_datetime(df["Date"])

# Set Date as index
df.set_index("Date", inplace=True)

# Check again after conversion
print(df.head(5))
print(df.info())

            Temp
Date            
1981-01-01  20.7
1981-01-02  17.9
1981-01-03  18.8
1981-01-04  14.6
1981-01-05  15.8
<class 'pandas.DataFrame'>
DatetimeIndex: 3650 entries, 1981-01-01 to 1990-12-31
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Temp    3650 non-null   float64
dtypes: float64(1)
memory usage: 57.0 KB
None
